src/js/utils/constants.js

```diff
@@ -1,19 +1,25 @@
 /**
  * Reusable constants
  */
-define([], function() {
+;(function(define) {
     'use strict';
-
-    return {
-        keyCodes: {
-            tab: 9,
-            enter: 13,
-            esc: 27,
-            space: 32,
-            left: 37,
-            up: 38,
-            right: 39,
-            down: 40
-        }
-    };
-});
+    define([], function() {
+        return {
+            keyCodes: {
+                tab: 9,
+                enter: 13,
+                esc: 27,
+                space: 32,
+                left: 37,
+                up: 38,
+                right: 39,
+                down: 40
+            }
+        };
+    });
+}).call(
+    this,
+    (typeof define !== 'undefined' && define) ||
```

Comment:
Thanks for the details about the syntax expected by the optimizer. I'll switch to that way of looking for ``define``. I don't understand why we don't need to use the ``RequireJS`` check though. This code is in the UI Toolkit and can be loaded in many different ways, including by code included in LMS factories where ``define`` is not defined and ``RequireJS.define`` is available. 

Let's discuss this in the morning and pin down right the combination of checks to be using in all these files. We might also want to recommend that LMS code use the same convention, rather than the current:

    define || RequireJS.define

I tried this originally here in the UI Toolkit but it doesn't work because ``define`` throws an undefined exception. The only reason it works in the LMS is due to this ugly hack where window is given a ``define`` property of undefined, rather than not having the property at all:

https://github.com/edx/edx-platform/blob/master/common/static/js/RequireJS-namespace-undefine.js#L7

For anyone reading along, here's an example of the standard idiom in the LMS:

https://github.com/edx/edx-platform/blob/master/lms/djangoapps/teams/static/teams/js/views/teams.js#L32